rubrics/api/src/main/java/org/sakaiproject/rubrics/logic/listener/WeightedEnabledListener.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2003-2021 The Apereo Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *             http://opensource.org/licenses/ecl2
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sakaiproject.rubrics.logic.listener;
+
+import javax.persistence.PostLoad;
+import org.sakaiproject.component.cover.ServerConfigurationService;
+
+import org.sakaiproject.rubrics.logic.model.Rubric;
+
+
+public class WeightedEnabledListener {
+
+    @PostLoad
+    public void isWeightedEnabled(Rubric rubric) {
+        boolean weightedEnabled = ServerConfigurationService.getBoolean("rubrics.enable.enable", false);
+        if(!weightedEnabled){
+            rubric.setWeighted(false);
```

Comment:
I don't like the idea of messing with the actual entity bean like this. Surely it's better to just update your db and set the weighted flag to false if you decide to disable weighting? The entity bean should reflect the db state, unless you're using a transient field.

Reply:
I understand your point, but should the need to update the DB data be documented somewhere?